### DISCUSSION TITLE
Mfem installation on native windows

### DISCUSSION THREAD

**Discussion by hbadi** [mfem/mfem#2939] [General]
Hi,
Just discovered mfem, it's seems very powerfull

I'd like to compile mfem on a native windows environment (not cygwin/mingw/WLS) using the intel compiler under VS. In order to not loosing time on something not possible or very complex, here my questions : 

- Does anybody already achieved that ?
- When configuring mfem, it appears that petsc needs MPI and Hypre. Does mfem can be compiled with a sequential petsc/slepc ? I compiled petsc/slepc with unimpi. Is it a good idea to enable MPI and fixing NP=1 ?
- Does Hypre can be compiled in native windows environment ?

Thanks you for helping.

**Comment by v-dobrev** [comment]
Hi @hbadi,

Sorry for the late reply.

As you noticed, to use Hypre and PETSc with MFEM you need to build it with MPI. Of course, using such a build on one MPI task should be totally fine.

Regarding native builds of Hypre and MFEM on windows: we test that with Visual Studio 2017 in our Appveyor CI, see https://github.com/mfem/mfem/blob/master/.appveyor.yml for the exact commands used for building there. The recommendations in https://github.com/mfem/mfem/issues/2834 may also be useful.

I'm not sure what exactly will be different when you use the Intel compiler with Intel's MPI library -- I have not tried that myself.

Let us know what you find out.
